packages/builder/src/package.ts

```diff
@@ -89,6 +89,17 @@ export class PackageReference {
   }
 
   static isValid(ref: string) {
+    const pkgRef = new PackageReference(ref);
```

Comment:
I added the error logic here so that I could run this function on the zod schema as a refine, however it feels wrong to add error logic to a boolean function. There's likely a better way to achieve this.

Reply:
Refactored this to add the error logic in the parse function of the package ref class, and add custom refinements in zod on the schema itself.